fail2ban/server/filter.py

```diff
@@ -518,19 +526,45 @@ def findFailure(self, tupleLine, date=None, returnRawHost=False,
 						 % ("\n".join(failRegex.getMatchedLines()), timeText))
 				else:
 					self.__lineBuffer = failRegex.getUnmatchedTupleLines()
+					# retrieve failure-id, host, etc from failure match:
+					raw = returnRawHost
```

Comment:
a bit confused here with initializing 'raw' with returnRawHost... so if that one is True (return raw), the at line 553, een if no host address was specified (so you get dns), you will then at 554 call IPAddr  on that fid.  Sounds incorrect.  Note that lines 547 - 550  weren't covered by test (codecov extension rules) that is why may be you didn't trigger this behaviour